### ISSUE TITLE
DOCX to HTML Heading Hierachy Breaks with Preceding List Item

### ISSUE THREAD

**Issue by thekenshow** [jgm/pandoc#5391]
I'm generating HTML from DOCX, and have run into an issue. When a section (created from DOCX heading levels) ends with a bulleted list item, the heading for the next section gets folded under the previous one as part of the bulleted list.

I can reproduce this using with:
```
pandoc -s "../Word/SourceDocument.docx" --number-sections -t html5 -o OuputDocument.html
```
Attached is a document that displays this problem.

[Pandoc Heading Structure Issue.docx](https://github.com/jgm/pandoc/files/2999324/Pandoc.Heading.Structure.Issue.docx)

**Comment by jkr** [comment]
There are actually two issues here:

1. The header being read as a single-item enumerated list at a level below the top (document) level (this can be solved by calling `walk` on `singleItemHeaderToHeader` instead of `map`).

2. But the second issue is why the header is being read at a lower level in the first place, since it seems on the page to be on the document level. I think this is because the attached document uses a bullet paragraph class (`BodyTextList`) instead of the stock list functionality. Given how obscure listing already is, this adds another layer of complication, and it's a bit hard to tell at first glance what the correct nesting really is. (Since the custom class changes the indenting, which is the main visual clue.)

In other words, though (1) seems like a good change, I'm not sure it would fix the problem in @thekenshow's file, since the heading-in-the-list might actually be what the document specifiies due to (2) -- even though it isn't visually evident.
